### PR TITLE
Custom events for tracking new filter UI interactions

### PR DESCRIPTION
**Pull request by samwinslow**
## Changes

In order to measure usage of the new filter UI, send some custom events.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

**Review by kpthatsme — COMMENTED** [review]
Hey @samwinslow, we keep custom event code on the frontend in this file https://github.com/PostHog/posthog/blob/master/frontend/src/lib/utils/eventUsageLogic.ts

Basically we create an action and reducer in that logic and then everywhere you're calling `posthog.capture` directly right now, you'd instead call the action.

**Comment by samwinslow** [comment]
This fails with some cryptic errors.

In the Trends component test: ` [KEA] Logic "scenes.PreflightCheck.logic", selector "socialAuthAvailable" has incorrect input: [undefined].` (preflightLogic should not be affected at all here).

It is very unclear to me when it is acceptable to call `someLogic.actions.actionName()` or not. The logic has to be mounted, which happens automatically in React but is very confusing otherwise.

Any advice? @paolodamico is out for another day but I'd love his input; seems to be the case that he has been able to call actions from outside React in `frontend/src/scenes/organization/Settings/invitesLogic.tsx:50`: `eventUsageLogic.actions.reportInviteAttempted( ...` There is no `mount()` call there, so why does it work?

Would just love to get unblocked on this ASAP even if it means refactoring later this week — don't want to miss the chance to start getting data.

**Comment by kpthatsme** [comment]
> This fails with some cryptic errors.
> 
> In the Trends component test: ` [KEA] Logic "scenes.PreflightCheck.logic", selector "socialAuthAvailable" has incorrect input: [undefined].` (preflightLogic should not be affected at all here).
> 
> It is very unclear to me when it is acceptable to call `someLogic.actions.actionName()` or not. The logic has to be mounted, which happens automatically in React but is very confusing otherwise.
> 

General rule of thumb is if where you're calling it from is a functional component, you use useActions in the component scope, so something like:

```
const { reportThisEvent } = useActions(eventUsageLogic)

<Component onClick=()=>{reportThisEvent()} />
```



> Any advice? @paolodamico is out for another day but I'd love his input; seems to be the case that he has been able to call actions from outside React in `frontend/src/scenes/organization/Settings/invitesLogic.tsx:50`: `eventUsageLogic.actions.reportInviteAttempted( ...` There is no `mount()` call there, so why does it work?
> 

In this case, it's being used in another logic, so we can't use the Kea hooks since those only work if you're in a functional component scope.

re: mounting – iirc once a logic is mounted, it remains mounted globally until unmounted.


> Would just love to get unblocked on this ASAP even if it means refactoring later this week — don't want to miss the chance to start getting data.

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 1]
```diff
@@ -362,5 +385,35 @@ export const eventUsageLogic = kea<
...
+        reportInsightFilterPropertyUpdated: ({ index }) => {
+            posthog.capture('filter property updated', { index })
+        },
+        reportInsightFilterMathUpdated: ({ index, math, mathProperty }) => {
+            posthog.capture('filter math updated', { math, mathProperty, index })
+        },
+        reportInsightFilterRemoved: ({ index }) => {
+            posthog.capture('local filter removed', { index })
+        },
+        reportInsightFilterAdded: ({ newLength }) => {
```
Comment (kpthatsme) [inline frontend/src/lib/utils/eventUsageLogic.ts:400]: Looks like you're missing the `async` keyword before the func defs

File: frontend/src/scenes/insights/InsightTabs/InsightActionBar.tsx [hunk 2]
```diff
@@ -22,6 +23,7 @@ export function InsightActionBar({ filters, annotations, insight, onReset }: Pro
                 onConfirm={() => {
                     window.scrollTo({ top: 0 })
                     onReset ? onReset() : push(`/insights?insight=${insight}`)
+                    posthog.capture('insights tab reset')
```
Comment (kpthatsme) [inline frontend/src/scenes/insights/InsightTabs/InsightActionBar.tsx:26]: Here we should be able to just call + import the eventUsagelogic action via the useActions hook

**Comment by samwinslow** [comment]
Failing tests unrelated; merging.

**Pull request merged**